tls-codec/tls_codec/src/arrays.rs

```diff
@@ -0,0 +1,45 @@
+//! Implement the TLS codec for some byte arrays.
+
+use super::{Deserialize, Error, Serialize, Size};
+use std::io::{Read, Write};
+
+macro_rules! impl_array {
+    ($($len:literal),*) => {
+        $(
+            impl Serialize for [u8; $len] {
+                fn tls_serialize<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+                    let written = writer.write(self)?;
+                    if written == $len {
+                        Ok(written)
+                    } else {
+                        Err(Error::InvalidWriteLength(format!("Expected to write {} bytes but only {} were written.", $len, written)))
+                    }
+                }
+            }
+
+            impl Deserialize for [u8; $len] {
+                fn tls_deserialize<R: Read>(bytes: &mut R) -> Result<Self, Error> {
+                    let mut out = [0u8; $len];
+                    bytes.read_exact(&mut out)?;
+                    Ok(out)
+                }
+            }
+
+            impl Size for [u8; $len] {
+                #[inline]
+                fn tls_serialized_len(&self) -> usize {
+                    $len
+                }
+            }
+        )*
+    };
+}
+
+impl_array!(
+    1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
+    27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50,
+    51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74,
+    75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98,
+    99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
+    118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128
+);
```

Comment:
Seems like something which would nicely be replaced with const generics, whenever you're ready to bump MSRV to 1.51.

Reply:
Indeed, we should replace this with const generics. I always forget that they are actually a thing now.